django-verdant/rca/navigation.py

```diff
@@ -0,0 +1,111 @@
+"""Provides the navigation data by requesting it from the new API.
+
+The new rebuild site is where the menu is being managed and updated.
+Instead of copying and pasting rendered html when the new site's menu gets
+updated, this template tag will request navigation data from the new site's
+navigation API endpoint. Instead of making this API call on every page request
+it's stored in cache['navigation_from_api'].
+
+"""
+
+
+import logging
+import requests
+from requests.auth import HTTPBasicAuth
+
+from django.conf import settings
+from django.core.cache import cache
+
+
+
+class CantPullFromRcaApi(Exception):
+    pass
+
+
+class Navigation(object):
+    def __init__(self, *args):
+        self.logger = logging.getLogger("verdant-rca")
+
+
+    def pull_navigation_data(self):
```

Comment:
Minor pedantic point but I think I prefer the verb `fetch` over `pull`, but feel free to ignore :-)